### PR TITLE
CHORE always introspect the subgraph, not the (federated) graph

### PR DESCRIPTION
**Pull request by cantremember**
this is what was causing the `yarn schema:push` to not work

a lot of stuff got introduced during the research effort -- it's good to keep it around, because the is *not the first* Apollo federated graph research effort we've had to do

i ended up re-building 'development' from scratch

### CODE REVIEW

File: script/apolloSchema.sh [hunk 1]
```diff
@@ -25,45 +25,67 @@ case $APOLLO_VARIANT in
     fi
     APOLLO_VARIANT='local'
     APOLLO_ENDPOINT_URL="http://localhost:$PORT/graphql"
-    APOLLO_VARIANT_CURRENT='development'
+    APOLLO_VARIANT_REMOTE='development'
```
Comment (cantremember) [inline script/apolloSchema.sh:28]: i think this is a clearer description of the purpose it serves

File: script/apolloSchema.sh [hunk 2]
```diff
@@ -25,45 +25,67 @@ case $APOLLO_VARIANT in
...
 function schemaDiff() {
-  $ROVER_PATH subgraph introspect $APOLLO_ENDPOINT_URL | APOLLO_KEY=$APOLLO_KEY $ROVER_PATH subgraph check joy-api-gateway@$APOLLO_VARIANT_CURRENT --name $APOLLO_SERVICE_NAME --schema -
+  schemaPreview | APOLLO_KEY=$APOLLO_KEY $ROVER_PATH subgraph check joy-api-gateway@$APOLLO_VARIANT_REMOTE --name $APOLLO_SERVICE_NAME --schema -
 }
 
+# update the Schema published to Apollo
 function schemaPush() {
   # how Stitch speaks to its peer Pods
   APOLLO_ROUTING_URL="http://bliss-$APOLLO_SERVICE_NAME:9000/graphql"
-  $ROVER_PATH graph introspect $APOLLO_ENDPOINT_URL | APOLLO_KEY=$APOLLO_KEY $ROVER_PATH subgraph publish joy-api-gateway@$APOLLO_VARIANT_CURRENT --name $APOLLO_SERVICE_NAME --routing-url $APOLLO_ROUTING_URL --schema -
```
Comment (cantremember) [inline script/apolloSchema.sh:58]: it's not obvious from the diffs -- but things suddenly started working when i consistently applied `schemaPreview` -- only later did i notice that that we were trying to jam the *entire Graph* into a Service's federated slice of the pie

hence the "only once" and "duplicate" failure messages

File: script/apolloSchema.sh [hunk 3]
```diff
@@ -25,45 +25,67 @@ case $APOLLO_VARIANT in
...
 
+# view the Schema currently published to Apollo
+function schemaPublished() {
+  $ROVER_PATH subgraph fetch joy-api-gateway@$APOLLO_VARIANT_REMOTE --name $APOLLO_SERVICE_NAME
+}
+
+# delete the Schema, to start from scratch
+function schemaDelete() {
+  APOLLO_KEY=$APOLLO_KEY $ROVER_PATH subgraph delete joy-api-gateway@$APOLLO_VARIANT_REMOTE --name $APOLLO_SERVICE_NAME
+}
```
Comment (cantremember) [inline script/apolloSchema.sh:77]: these are useful.  but there's no reason to turn them into `yarn` tasks

**Review by vikas-mindpath — APPROVED** [review]
**lgtm**

**Review by tomnocon — APPROVED** [review]

**Pull request merged**